From d4f19c5cd2e36324176168212f9cf70ebbf5af5d Mon Sep 17 00:00:00 2001
From: Tifa Chen <tifa.chen@technexion.com>
Date: Tue, 31 Mar 2026 17:26:14 +0800
Subject: [PATCH] Using videorate to specific camera framerate

Delete the set camera framerate to pipline and change to use video
output a specific frame amount. Make it more flexible when using
high-spec cameras.
---
 common/cpp/src/gst_source_imx.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/common/cpp/src/gst_source_imx.cpp b/common/cpp/src/gst_source_imx.cpp
index 1f950d1..b003630 100644
--- a/common/cpp/src/gst_source_imx.cpp
+++ b/common/cpp/src/gst_source_imx.cpp
@@ -53,7 +53,7 @@ void GstCameraImx::addCameraToPipeline(GstPipelineImx &pipeline)
   cmd += "v4l2src name=" + gstName + " device=" + device.string();
   cmd += " num-buffers=-1 ! video/x-raw,width=";
   cmd += std::to_string(width) + ",height=" + std::to_string(height);
-  cmd += ",framerate=" + std::to_string(framerate) + "/1 ! ";
+  cmd += " ! videorate ! video/x-raw,framerate=30/1 ! ";
   pipeline.addToPipeline(cmd);
 
   if ((format.length() != 0) or flip)
-- 
2.25.1

